operator/controller/src/main/java/io/apicurio/registry/operator/utils/PodTemplateSpecUtils.java

```diff
@@ -0,0 +1,147 @@
+     * Merge the original PTS from the factory into a copy of the PTS from the spec.
+     *
+     * @return merged copy
+     */
+    public static PodTemplateSpec mergePTS(PodTemplateSpec spec, PodTemplateSpec original,
+            String containerName) {
+
+        requireNonNull(original);
+        if (spec == null) {
+            return original;
+        }
+
+        var merged = duplicate(spec, PodTemplateSpec.class);
+
+        // .metadata
+        if (merged.getMetadata() == null) {
+            merged.setMetadata(new ObjectMeta());
+        }
+
+        // .metadata.labels
+        if (merged.getMetadata().getLabels() == null) {
+            merged.getMetadata().setLabels(new HashMap<>());
+        }
+        if (original.getMetadata() != null) {
+            mergeOverride(merged.getMetadata().getLabels(), original.getMetadata().getLabels());
+        }
+
+        // .metadata.annotations
+        if (merged.getMetadata().getAnnotations() == null) {
+            merged.getMetadata().setAnnotations(new HashMap<>());
+        }
+        if (original.getMetadata() != null) {
+            mergeOverride(merged.getMetadata().getAnnotations(), original.getMetadata().getAnnotations());
+        }
+
+        // .spec.containers[name = containerName]
+        var oc = getContainer(original, containerName);
+        if (oc == null) {
+            throw new OperatorException("Container %s not found in the original PTS:\n%s"
+                    .formatted(containerName, toYAML(original)));
+        }
+
+        // .spec
+        if (merged.getSpec() == null) {
+            merged.setSpec(new PodSpec());
+        }
+
+        // .spec.containers
+        if (merged.getSpec().getContainers() == null) {
+            merged.getSpec().setContainers(new ArrayList<>());
+        }
+
+        var mc = getContainer(merged, containerName);
+        if (mc == null) {
+            merged.getSpec().getContainers().add(oc);
+        } else {
+
+            // .spec.containers[name = containerName].env
+            if (!isEmpty(mc.getEnv())) {
+                throw new OperatorException("""
+                        Field spec.(app/ui).podTemplateSpec.spec.containers[name = %s].env must be empty.  \
+                        Use spec.(app/ui).env to configure environment variables."""
+                        .formatted(containerName));
+            }
+            mc.setEnv(oc.getEnv());
+
+            // .spec.containers[name = containerName].image
+            if (!isEmpty(mc.getImage())) {
+                // TODO: Do we want to allow this?
+                throw new OperatorException("""
+                        Field spec.(app/ui).podTemplateSpec.spec.containers[name = %s].image must be empty."""
+                        .formatted(containerName));
+            }
+            mc.setImage(oc.getImage());
+
+            // .spec.containers[name = containerName].ports
+            mergeNotOverride(mc.getPorts(), oc.getPorts(), ContainerPort::getName);
```

Comment:
This method is complex enough to keep

Reply:
it's a one liner:

```java
source.forEach(target::putIfAbsent);
```

@jsenko please, let me "Resolve" the conversations, otherwise is really hard for me to validate the progress :pray:

Reply:
oh, this is the overload :sweat_smile: let me have a closer look.

Reply:
Ok, the implementation is something like (not tested):

```java
       source.forEach((sKey, sValue) -> {
            target.forEach((tKey, tValue) -> {
                if (sValue.getName().equals(tValue.getName())) {
                    target.put(sKey, sValue);
                };
            });
        });
```

Doing it I realized that I'm not sure about the semantics, here we are using the name of the port, I think that we should use the key of the map and document the behavior accordingly.
The rationale is:
looking inside the `ContainerPort` object one can argue why we are using the name and not the port number, and we are not considering the other fields(like protocol etc.).
The outern key, instead, is a direct identifier that cannot be duplicated due to yaml not accepting it.

Bonus point:
the current implementation is computationally complex as it does full target table scans, one each source Map.
I don't expect a lot of elements in those maps, so this just a nice side effect.